presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -137,10 +139,14 @@
     private long lastStartMillis;
     @GuardedBy("root")
     private final CounterStat timeBetweenStartsSec = new CounterStat();
+    @GuardedBy("root")
+    private final int version;
```

Comment:
If this is final and it's a primitive type, why does it need to be synchronized?

Reply:
it's for the older code, with new changes we don't need it anymore. removing  the synchronization part.